src/lib/auth/auth.ts

```diff
@@ -0,0 +1,127 @@
+/* Facade for keycloak */
+import Keycloak, { KeycloakInitOptions, KeycloakInstance } from 'keycloak-js';
+
+const toPromise = keycloakPromise =>
+  new Promise<any>((resolve, reject) => keycloakPromise.success(resolve).error(reject));
+
+export interface ResourceRole {
+  resource: string;
+  resourceId: string;
+  role: string;
+}
+
+export interface AuthServiceConfiguration {
+  oidcIssuer: string;
+  clientId: string;
+  redirectUri: string;
+  logoutRedirectUri: string;
+}
+
+export interface User {
+  username: string;
+  name: string;
+}
+
+const extractUser: (string) => User = tokenParsed => ({ username: tokenParsed.user_name, name: tokenParsed.name });
+
+const extractResourceRoles: (authorities: string) => ResourceRole[] = (authorities: string) => {
+  return (authorities || '')
+    .split(',')
+    .map(authorityDescriptor => authorityDescriptor.split(':'))
+    .map(parts => ({ resource: parts[0], resourceId: parts[1], role: parts[2] }));
+};
+
+export class Auth {
+  private readonly kc: KeycloakInstance;
+
+  private readonly conf: AuthServiceConfiguration;
+
+  private initialized = false;
+
+  constructor(conf: AuthServiceConfiguration) {
+    const [url, realm] = conf.oidcIssuer.split('/realms/');
+
+    const kcConfig = {
+      realm,
+      url,
+      clientId: conf.clientId
+    };
+
+    this.conf = conf;
+    this.kc = Keycloak(kcConfig);
+  }
+
+  async init({ loginRequired }: { loginRequired: boolean }): Promise<boolean> {
```

Comment:
Do not declare inline interfaces, rather declare AuthServiceInterface separately

Reply:
så jeg prøver det, men det ser ikke ut at det setter noe restriksjoner i implementasjon. vi kan se det sammen kanskje.